src/main/java/jenkins/plugins/logstash/persistence/LogzioDao.java

```diff
@@ -0,0 +1,127 @@
+package jenkins.plugins.logstash.persistence;
+
+import com.github.wnameless.json.flattener.JsonFlattener;
+import com.google.common.io.Files;
+
+import io.logz.sender.LogzioSender;
+import io.logz.sender.SenderStatusReporter;
+import io.logz.sender.com.google.gson.JsonObject;
+import io.logz.sender.exceptions.LogzioParameterErrorException;
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import jenkins.model.Jenkins;
+import jenkins.plugins.logstash.LogstashConfiguration;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+/**
+ * Logz.io Data Access Object.
+ *
+ * @author Ido Halevi
+ */
+
+public class LogzioDao extends AbstractLogstashIndexerDao {
+    private static final int CONNECT_TIMEOUT = 10*1000;
+    private static final int CORE_POOL_SIZE = 2;
+    private static final int DRAIN_TIMEOUT = 2;
+    private static final int FS_PERCENT_THRESHOLD = 99;
+    private static final int GC_PERSISTED_QUEUE_FILE_INTERVAL_SECOND = 30;
+    private static final int SOCKET_TIMEOUT = 10*1000;
+    private static final String TYPE = "jenkins_plugin";
+    private final LogzioSender logzioSender;
+
+    private String key;
+    private String host;
+
+    //primary constructor used by indexer factory
+    public LogzioDao(String host, String key){
+        this(null, host, key);
+    }
+
+    // Factored for unit testing
+    LogzioDao(LogzioSender factory, String host, String key){
+        this.host = host;
+        this.key = key;
+
+        // create file for sender queue
+        File fp;
+        Jenkins jenkins = Jenkins.getInstanceOrNull();
+        if (jenkins == null){
+            // for testing (mvn package)
+            fp = Files.createTempDir();
+        }else{
+            fp = new File(jenkins.getRootDir() + "/logs/logzio_plugin");
+            boolean folderExisted = fp.mkdirs() || fp.setWritable(true);
+            if (!folderExisted) {
+                throw new IllegalArgumentException("Unable to create path");
+            }
+        }
+
+        try{
+            this.logzioSender = factory == null ? LogzioSender.getOrCreateSenderByType(key, TYPE, DRAIN_TIMEOUT,
+                    FS_PERCENT_THRESHOLD, fp, host, SOCKET_TIMEOUT, CONNECT_TIMEOUT,false, null,
+                    Executors.newScheduledThreadPool(CORE_POOL_SIZE),GC_PERSISTED_QUEUE_FILE_INTERVAL_SECOND, true) : factory;
+            this.logzioSender.start();
+        }catch (LogzioParameterErrorException e){
+            throw new IllegalArgumentException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void push(String data) {
+        JSONObject jsonData = JSONObject.fromObject(data);
+        JSONArray logMessages = jsonData.getJSONArray("message");
+        for (Object logMsg : logMessages) {
+            JsonObject logLine = createLogLine(jsonData, logMsg.toString());
```

Comment:
You're splitting the messages when sent from the Notifier. Would it make sense to make this a general feature?

Reply:
I'd rather work on transforming the data as part of #71

Reply:
I think it should be a general feature, one of the benefits of ES and Logz.io is the visualization of the logs, for that we need to have each log separately. Also in a big log, the HTTP/HTTPs request might be too big no?

Reply:
The bulk send is only used in the notifier and logstashSend pipeline step. When using the other way it is line by line sending.

Reply:
I don't understand why we would want to split the lines here. If you want line by line then just use the Wrapper instead of Notifier as @mwinter69  is suggesting

Reply:
@jakub-bochenski 
I understand your answer. This came from my testing with small log file (20MB), I noticed that when we use the Notifier it is X7 faster than the Wrapper, so whoever uses the Wrapper will suffer from a much longer build time. If you don't think it's a big issue I'll remove it

Reply:
I've never noticed such an overhead of notifier vs wrapper. Can you post the details of your test?

Reply:
I think the overhead factor comes from the fact that you are generating a lot of output in little time. Usually running tests or even compiling has a much higher processing time/text output ratio.

I would leave the processing as it was before for now (wrapper = per line, notifier = single document). If you still want this I'd do that in a separate PR, preferably covering this on the notifier level to cover all daos.